sml/metrics/classification/classification.py

```diff
@@ -16,13 +16,92 @@
 
 import jax
 import jax.numpy as jnp
+import numpy as np
 
 from sml.preprocessing.preprocessing import label_binarize
 from spu.ops.groupby import groupby, groupby_sum
 
 from .auc import binary_clf_curve, binary_roc_auc
 
 
+def confusion_matrix(y_true, y_pred, labels, sample_weight=None, normalize=None):
+    """calculate the confusion matrix"""
+    y_true = jnp.array(y_true)
+    y_pred = jnp.array(y_pred)
+
+    # 获取标签的数量
+    num_labels = len(labels)
+
+    # 初始化混淆矩阵
+    cm = jnp.zeros((num_labels, num_labels), dtype=jnp.int32)
+
+    # 计算混淆矩阵
```

Comment:
Please use English for comments, keep consistent to the project style.

Reply:
Update code comments with English.